commons/src/main/java/org/infinispan/commons/util/Util.java

```diff
@@ -475,6 +476,13 @@ public static String toStr(Object o) {
          return printArray((byte[]) o, false);
       } else if (o == null) {
          return "null";
+      } else if (o.getClass().isArray()) {
```

Comment:
If I see correctly, `printArray` handles only `byte[]` and this was already handled earlier in this if-else switch. This variant is for generic arrays. I could refactor that to `printArray(Object)`, if you want, but we can't even use `printArray(T[])` since we don't know the type of array statically.